beginend.el

```diff
@@ -232,6 +232,32 @@ BEGIN-BODY and END-BODY are two `progn' expressions passed to respectively
   (progn
     (prodigy-last)))
 
+;; Copy/pasted from expand-region
+(defun beginend--point-is-in-comment-p ()
+  "Return non-nil if point is in comment."
+  (or (nth 4 (syntax-ppss))
+      (memq (get-text-property (point) 'face)
+            '(font-lock-comment-face font-lock-comment-delimiter-face))))
+
+(defun beginend--prog-mode-code-position-p ()
+  "Return non-nil if point, at beginning of line, is inside code."
+  (not
+   (or (beginend--point-is-in-comment-p)
+       (eq (point) (line-end-position))
```

Comment:
I would use `=` over `eq` for numbers (`eq` generally is "reserved" for symbols).

But why is the check needed?  Wouldn't it be enough to jump to the beginning and test for the comment?

I'm thinking of a check like this:

1. save excursion
2. go to the beg of line
3. skip whitespace
4. decide comment status
    * if syntax-ppss on 4th element returns non-nil, we must be in comment, if not check the following
    * if looking at a comment starter -> we are at a comment-only line